torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3546,8 +3546,22 @@ struct to_ir {
           range, sv->asValue(val_range, method), subscript_exprs));
     }
     if (subscript_exprs[0].kind() == TK_SLICE_EXPR) {
-      return std::make_shared<SimpleValue>(emitBasicSlice(
-          range, sv->asValue(val_range, method), subscript_exprs));
+      if(sv->kind() == "module") {
+        // TODO Special case for Sequential only, make sure not other modules
```

Comment:
@eellison Do you have any thoughts on how to restrict this case better so that it only applies to Sequentials?  The only thing I could think of doing was strcmp between sv->asValue(val_range, method)->type()->str() and "__torch__.torch.nn.modules.container.Sequential".  Incidentally, this passes all of the tests as is (test_jit at least).

Reply:
I think we can just put the error checking for non-Sequentials/ModuleLists in the `asTuple` method of the PythonModule.

Checking for the kind() string seems a little bit like an anti-pattern but i'm not what's better here honestly. normally we could dynamic cast check for the type of the sugared value but we can't do that here...

Reply:
yea, i tried the cast route earlier, couldn't come up with anything smarter here. not my proudest line of code.

Reply:
I guess the semantically equivalent but maybe more principled thing here to do would be to add an api that all SugaredValues have to define that returns a Symbol. Just as we have SpecialFormValue that is instantiated with a Symbol and then that Symbol is dispatched on in ir_emitter.cpp

However this is semantically the same, probably better as a follow up since as I mentioned this would PR would be great to have in the release.